### PR TITLE
use log funcs of core framework in the auth, kubectl and windows pkg

### PR DESCRIPTION
**Pull request by carlory**
**What type of PR is this?**

/kind cleanup
/kind testing

**What this PR does / why we need it**:

use log funcs of core framework

```
test/e2e/auth/*.go
test/e2e/windows/*.go
test/e2e/kubectl/*.go
```
**Which issue(s) this PR fixes**:

refer #81427

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

```release-note
NONE
```

**Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.**:

```docs
NONE
```

### CODE REVIEW

**Review by oomichi — COMMENTED** [review]
/assign @oomichi

**Comment by carlory** [comment]
@oomichi fixed

File: test/e2e/auth/metadata_concealment.go [hunk 1]
```diff
@@ -22,9 +22,9 @@ import (
 	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
 	"k8s.io/kubernetes/test/e2e/framework"
 	jobutil "k8s.io/kubernetes/test/e2e/framework/job"
+	imageutil "k8s.io/kubernetes/test/utils/image"
 
 	"github.com/onsi/ginkgo"
-	imageutil "k8s.io/kubernetes/test/utils/image"
```
Comment (oomichi) [inline test/e2e/auth/metadata_concealment.go]: unrelated change in this PR.

File: test/e2e/auth/node_authn.go [hunk 2]
```diff
@@ -23,11 +23,11 @@ import (
 	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
 	"k8s.io/kubernetes/pkg/master/ports"
 	"k8s.io/kubernetes/test/e2e/framework"
+	e2enode "k8s.io/kubernetes/test/e2e/framework/node"
 	imageutils "k8s.io/kubernetes/test/utils/image"
 
 	"github.com/onsi/ginkgo"
 	"github.com/onsi/gomega"
-	e2enode "k8s.io/kubernetes/test/e2e/framework/node"
```
Comment (oomichi) [inline test/e2e/auth/node_authn.go]: ditto

File: test/e2e/windows/BUILD [hunk 3]
```diff
@@ -32,7 +32,6 @@ go_library(
         "//staging/src/k8s.io/kubelet/config/v1beta1:go_default_library",
         "//test/e2e/framework:go_default_library",
         "//test/e2e/framework/kubelet:go_default_library",
-        "//test/e2e/framework/log:go_default_library",
```
Comment (oomichi) [inline test/e2e/windows/BUILD]: The CI job `pull-kubernetes-bazel-build` continues to be failed because of
```
compile: missing strict dependencies:
	/b/f/w/test/e2e/windows/dns.go: import of "k8s.io/kubernetes/test/e2e/framework/log"
	/b/f/w/test/e2e/windows/security_context.go: import of "k8s.io/kubernetes/test/e2e/framework/log"
```
That means these go files still use e2elog but this PR removes it from BUILD file.
So we need to take care of these files also in this PR.

Comment (carlory) [reply test/e2e/windows/BUILD]: I cannot find the `dns.go` and `security_context.go` in the `test/e2e/windows` pkg

```
windows
├── BUILD
├── OWNERS
├── README.md
├── density.go
├── framework.go
├── gmsa.go
├── hybrid_network.go
├── memory_limits.go
├── networking.go
├── service.go
└── volumes.go

0 directories, 11 files
```

**Comment by oomichi** [comment]
We need to remove e2elog usage from 3 files also (dns.go, gmsa_full.go and security_context.go) under test/e2e/windows/ as

```
compile: missing strict dependencies:
	/b/f/w/test/e2e/windows/dns.go: import of "k8s.io/kubernetes/test/e2e/framework/log"
	/b/f/w/test/e2e/windows/gmsa_full.go: import of "k8s.io/kubernetes/test/e2e/framework/log"
	/b/f/w/test/e2e/windows/security_context.go: import of "k8s.io/kubernetes/test/e2e/framework/log"
```
as https://storage.googleapis.com/kubernetes-jenkins/pr-logs/pull/81987/pull-kubernetes-bazel-build/1168008132962553856/build-log.txt

**Review by oomichi — COMMENTED** [review]
Thanks for updating

/lgtm
/approve

**Comment by carlory** [comment]
@oomichi 

it is ready for re-review.

**Review by oomichi — COMMENTED** [review]
/lgtm

**Review by feiskyer — COMMENTED** [review]
/lgtm
/retest
